src/runtime/pkg/containerd-shim-v2/shim_management.go

```diff
@@ -126,6 +138,47 @@ func decodeAgentMetrics(body string) []*dto.MetricFamily {
 	return list
 }
 
+func (s *service) serveVolumeStats(w http.ResponseWriter, r *http.Request) {
+	volumePath, err := url.PathUnescape(strings.TrimPrefix(r.URL.Path, DirectVolumeStatUrl))
+	if err != nil {
+		shimMgtLog.WithError(err).Error("failed to unescape the volume stat url path")
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(err.Error()))
+	}
```

Comment:
Aren't all these error blocks missing a `return`?

Reply:
@jodh-intel Thanks for catching this! I don't know how I would miss it :(